cpp/src/Ice/LocalExceptions.cpp

```diff
@@ -0,0 +1,948 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#include "Ice/LocalExceptions.h"
+#include "Ice/Initialize.h"
+#include "Ice/StringUtil.h"
+#include "Network.h"
+#include "RequestFailedMessage.h"
+
+#include <iomanip>
+
+using namespace std;
+using namespace Ice;
+using namespace IceInternal;
+
+namespace
+{
+    inline string socketErrorToString(int error)
+    {
+        if (error == 0)
+        {
+            return "unknown error";
+        }
+        return IceInternal::errorToString(error);
+    }
+}
+
+namespace
+{
+    inline std::string createRequestFailedMessage(const char* typeName)
+    {
+        ostringstream os;
+        os << "dispatch failed with " << typeName;
+        return os.str();
+    }
+}
+
+// Can't move id/facet/operation because the evaluation order is unspecified.
+// https://en.cppreference.com/w/cpp/language/eval_order
+Ice::ObjectNotExistException::ObjectNotExistException(
+    const char* file,
+    int line,
+    Identity id,
+    string facet,
+    string operation)
+    : RequestFailedException(
+          file,
+          line,
+          createRequestFailedMessage("ObjectNotExistException", id, facet, operation),
+          id,
+          facet,
+          operation)
```

Comment:
can we call std::move for these parameters?

Reply:
See the "Can't move" comment on line 39 above. Am I missing something here?